### PR TITLE
CLOUD-579 - fix smart-update test

### PR DESCRIPTION
**Pull request by tplavcic**
[CLOUD-579](https://jira.percona.com/browse/CLOUD-579)

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| haproxy | passed |
| big-data | passed |
| smart-update | failed |
| recreate | passed |
| storage | passed |
| scaling | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| init-deploy | passed |
| upgrade | passed |
| restore-to-encrypted-cluster | passed |
| limits | passed |
| self-healing-advanced | passed |
| upgrade-consistency | passed |
| monitoring | passed |
| operator-self-healing | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| affinity | passed |
| security-context | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | failed |
| scheduled-backup | passed |
| demand-backup-encrypted-with-tls | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| haproxy | passed |
| smart-update | passed |
| storage | passed |
| recreate | passed |
| scaling | passed |
| big-data | passed |
| init-deploy | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| upgrade | failed |
| restore-to-encrypted-cluster | passed |
| upgrade-consistency | passed |
| limits | failed |
| self-healing-advanced | passed |
| demand-backup | passed |
| security-context | passed |
| monitoring | passed |
| operator-self-healing | failed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | failed |
| auto-tuning | passed |
| scheduled-backup | passed |
| proxysql-sidecar-res-limits | passed |
| users | failed |
| demand-backup-encrypted-with-tls | passed |

**Review by cap1984 — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| smart-update | passed |
| recreate | passed |
| haproxy | passed |
| restore-to-encrypted-cluster | passed |
| init-deploy | passed |
| demand-backup | passed |
| limits | failed |
| scaling | passed |
| scheduled-backup | passed |
| scaling-proxysql | passed |
| demand-backup-encrypted-with-tls | passed |
| upgrade | failed |
| storage | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | failed |
| big-data | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | failed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | failed |

File: e2e-tests/smart-update/run [hunk 1]
```diff
@@ -5,10 +5,28 @@ set -o xtrace
...
+
+if [[ "${TARGET_IMAGE_PXC}" == *"percona-xtradb-cluster-operator"* ]]; then
+    PXC_VER=$(echo -n "${TARGET_IMAGE_PXC}" | $sed -r 's/.*([0-9].[0-9])$/\1/')
+else
+    PXC_VER=$(echo -n "${TARGET_IMAGE_PXC}" | $sed -r 's/.*:([0-9]+\.[0-9]+).*$/\1/')
+fi
+if [[ "${PXC_VER}" == "5.7" ]]; then
+    IMAGE_PXC="percona/percona-xtradb-cluster:5.7.30-31.43"
+elif [[ "${PXC_VER}" == "8.0" ]]; then
+    IMAGE_PXC="percona/percona-xtradb-cluster:8.0.19-10.1"
```
Comment (hors) [inline e2e-tests/smart-update/run:23]: Why do we need to hardcode it? We need to have possibility to test images which we pass via test's arguments/options.

File: e2e-tests/smart-update/run [hunk 2]
```diff
@@ -64,134 +68,152 @@ function check_last_pod_to_update {
...
     kubectl_bin delete pvc --all
 
+    ##################################################
     desc 'PXC cluster update with explicitly specified image inside version service'
-    spinup_pxc "$cluster" "$test_dir/conf/$cluster-version-service-unreachable.yml"
-    version_service_image="8.0.19-10.1"
-    initial_primary=$(run_mysql 'SELECT @@hostname hostname;' "-h $cluster-haproxy -uroot -proot_password")
+    spinup_pxc "${CLUSTER}" "${tmp_dir}/${CLUSTER}-version-service-unreachable.yml"
+    if [[ "${PXC_VER}" == "5.7" ]]; then
+        vs_image="5.7.31-31.45"
```
Comment (hors) [inline e2e-tests/smart-update/run:196]: I do not like the idea that we hardcode it. Can we avoid it?

**Review by hors — CHANGES_REQUESTED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| smart-update | failed |
| storage | passed |
| big-data | passed |
| recreate | passed |
| scaling | passed |
| haproxy | passed |
| self-healing | passed |
| init-deploy | passed |
| scaling-proxysql | passed |
| restore-to-encrypted-cluster | passed |
| upgrade | failed |
| upgrade-consistency | passed |
| limits | failed |
| security-context | passed |
| self-healing-advanced | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| operator-self-healing | failed |
| affinity | passed |
| one-pod | failed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | failed |
| demand-backup | failed |
| scheduled-backup | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| haproxy | passed |
| recreate | passed |
| smart-update | failed |
| big-data | passed |
| storage | passed |
| init-deploy | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| upgrade | failed |
| restore-to-encrypted-cluster | passed |
| upgrade-consistency | passed |
| limits | failed |
| demand-backup | passed |
| security-context | passed |
| monitoring | passed |
| self-healing-advanced | passed |
| monitoring-2-0 | passed |
| operator-self-healing | failed |
| affinity | passed |
| one-pod | failed |
| auto-tuning | passed |
| scheduled-backup | passed |
| proxysql-sidecar-res-limits | passed |
| users | failed |
| demand-backup-encrypted-with-tls | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| storage | passed |
| haproxy | passed |
| big-data | passed |
| smart-update | passed |
| scaling | passed |
| init-deploy | passed |
| scaling-proxysql | passed |
| restore-to-encrypted-cluster | passed |
| self-healing | passed |
| upgrade | failed |
| upgrade-consistency | passed |
| limits | failed |
| self-healing-advanced | passed |
| demand-backup | passed |
| security-context | passed |
| monitoring | passed |
| operator-self-healing | failed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | failed |
| auto-tuning | passed |
| scheduled-backup | passed |
| proxysql-sidecar-res-limits | passed |
| users | failed |
| demand-backup-encrypted-with-tls | passed |

File: e2e-tests/smart-update/run (hunk 1 above)
Comment (tplavcic) [reply e2e-tests/smart-update/run]: These images are old version images from which we do upgrade to the images which are selected. We have a custom version service deployed and I have changed this to query the version service to get old images.

File: e2e-tests/smart-update/run (hunk 2 above)
Comment (tplavcic) [reply e2e-tests/smart-update/run]: Thanks! Fixed.

**Review by hors — APPROVED** [review]

**Review by hors — APPROVED** [review]

**Review by cap1984 — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| storage | passed |
| scaling | passed |
| upgrade-haproxy | failed |
| haproxy | passed |
| big-data | passed |
| init-deploy | passed |
| self-healing | passed |
| restore-to-encrypted-cluster | passed |
| upgrade-proxysql | failed |
| limits | failed |
| scaling-proxysql | passed |
| self-healing-advanced | passed |
| security-context | passed |
| smart-update | passed |
| demand-backup | passed |
| operator-self-healing | failed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | failed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| upgrade-consistency | passed |
| users | passed |
| scheduled-backup | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| demand-backup-encrypted-with-tls | failed |
| validation-hook | passed |

**Pull request merged**